### ISSUE TITLE
kpt cfg set fails on empty value

### ISSUE THREAD

**Issue by yhrn** [GoogleContainerTools/kpt#721]
If I try to set the empty string as a value with kpt 0.29.0 it fails with the error message:
```shell
> kpt-0.29.0 cfg set . name "" --set-by yhrn
set 0 fields
Error: invalid character '}' looking for beginning of valueUsage:
  set DIR NAME --values [VALUE] [flags] 

Examples:

...snip...

Error: invalid character '}' looking for beginning of value
```

**On kpt 0.26.0 this works as expected.**

My package looks like this:

Kptfile
```yaml
apiVersion: kpt.dev/v1alpha1
kind: Kptfile
metadata:
  name: setter-bug
packageMetadata:
  shortDescription: sample description
openAPI:
  definitions:
    io.k8s.cli.setters.name:
      description: reason
      x-k8s-cli:
        setter:
          name: name
          value: some-topic
          setBy: yhrn
```

topic.yaml
```yaml
apiVersion: pubsub.cnrm.cloud.google.com/v1beta1
kind: PubSubTopic
metadata:
  name: some-topic # {"$ref":"#/definitions/io.k8s.cli.setters.name"}
```

**Comment by mortent** [comment]
@yhrn Thanks for reporting this.

This is caused by incorrect handling of empty strings when we try to determine the type of the value. It was fixed in https://github.com/kubernetes-sigs/kustomize/pull/2570 and is included in kpt v0.30.0 that we released today.

**Comment by yhrn** [comment]
I can confirm that this is working as expected in 0.30.0
